### ISSUE TITLE
Thunderbird 78

### ISSUE THREAD

**Issue by dougbreaux** [jmozmoz/compactheader#42]
In case author or anyone else can get to trying to make this work on Thunderbird 78.

https://developer.thunderbird.net/add-ons/updating/tb78

**Comment by jmozmoz** [comment]
I have updated one of my much small addons: https://github.com/jmozmoz/autoslide It took me about the equivalent of one working day to get this done. (And actually, I found a bug in Thunderbird's core, which nobody seems to work on until now: https://bugzilla.mozilla.org/show_bug.cgi?id=1659854) Extrapolating this experience to CompactHeader boils down to several weeks of work.

Even, if I updated CompactHeader now, it would probably only be a solution for the current 78 ESR version and in one year, everything has to be done again. Also, there might even be much more changes to the user interface of Thunderbird: https://www.youtube.com/watch?v=pbhrssxFWzA

So to sum up: I think, this should be solved in the core of Thunderbird (https://bugzilla.mozilla.org/show_bug.cgi?id=634831). The core developers have to find a way how Thunderbird should work on smaller displays.

**Comment by eyalroz** [comment]
@jmozmoz : Hello Joachim. Extension  developer here ([removedupes](https://github.com/eyalroz/removedupes/tree/development), [bidimailui](https://github.com/eyalroz/bidimailui/tree/development)).

First, I agree that there should be action in the core of Thunderbird about this. However - the prospects of adapting your extension for TB 78 are more positive than you may be aware. You see, it actually _is_ possible to run regular extensions with full privileges in TB 78. What we're missing is just the mechanism to load them. Fortunately, an alternative regular-extension-loading mechanism has been implemented as a "Web Experiment". It's called WindowListener (but never mind the name), 

Essentially it replaces loading via `chrome.manifest` with some Javascript telling the loader which XUL documents you need to alter, where your prefs dialog is, etc.  [Example](https://github.com/eyalroz/removedupes/blob/development/src/background.js)

Also, instead of a proper overlay loader, we need to wrap our XUL with a bit of glue code: [Example](https://github.com/eyalroz/removedupes/blob/development/src/chrome/content/overlay-injectors/messenger.js).

So, this way of loading is somewhat uglier than before, but it certainly doesn't take weeks to implement. Your JS code does _not_ need to change (except as @dougbreaux linked to, which is small potatoes).

There is also a [detailed guide](https://github.com/thundernest/addon-developer-support/wiki/) about how to use the WindowListener experiment, what to put in `background.js` to it, how to arrange the injection of XUL elements etc. John Bieling wrote it and I made a bunch of suggestions to improve and clarify it, which he has.

Finally, it is important that we don't lose significant extensions (like yours). There is a tendency among some of the Thunderbird Council to dismiss the importance of extensions, and relegate us to using those crippled WebExtension APIs. If we drop our extensions, we will be making that a self-fulfilling prophecy - only trivial ones will remain. 

So - please consider updating your extension!

**Comment by jmozmoz** [comment]
As I wrote I went through the process of updating a really small addon. Most of the work were cleanups, like making sure, that all observers are removed if the addon is disabled/uninstalled. 

A lot of code in CompactHeader is also related to the header pane toolbar, which now cannot be customized anymore (but see https://addons.thunderbird.net/en-US/thunderbird/addon/msghdr-toolbar-customize/). 

So really a re-write/major cleanup of the CompactHeader code would be necessary. The next problem is css styles for different operating systems, ... 

I had a quite comprehensive test suite using mozmill, but this of course is also dead. So pushing a major rewrite to a lot of users (without a beta channel to test it) is also a big risk. I do not want to suddenly break Thunderbird for a few 1000s or 10000s users.

This was a pure hobby project which I started a long time ago, because at that time I had a  13" screen with 768(?) pixel height. But that laptop died a long time ago. 

I also do not want to get involved in all the troubles the Thunderbird project is going through. And if I understand [this](https://blog.mozilla.org/addons/2020/09/09/introducing-the-promoted-add-ons-pilot/) correctly Mozilla plans to take money from (Firefox) add-on developers for review. Guess how long it will take, until add-ons will only be allowed if you pay?

In the worst case, I have to setup a [roundcube](https://roundcube.net/) server and use my one web mail.

**Comment by alanmead** [comment]
> Even, if I updated CompactHeader now, it would probably only be a solution for the current 78 ESR version and in one year, everything has to be done again. Also, there might even be much more changes to the user interface of Thunderbird: https://www.youtube.com/watch?v=pbhrssxFWzA

Wow, this is really disappointing. That YT talk is so arrogant. He starts by saying that he gets negative feedback ("blowing things up") every time he makes a change and shortly before 13:00 he says users are wrong in their complaints because they don't get it. I stopped watching at that point. It doesn't matter how good his theory is, if he fails in practice.

And TB 78 is a lot less usable on a fairly large screen without CH. 

Also TB discouraging add-on devs (even if that's not intentional) seems like a big problem.

**Comment by eyalroz** [comment]
@alanmead : Can you write me an email please? I'd like to comment on what you said but not in relation to CompactHeader. My address is on my profile page.

**Comment by dougbreaux** [comment]
There is another similar addon that is working for me: https://addons.thunderbird.net/en-US/thunderbird/addon/compact-headers/

**Comment by dougbreaux** [comment]
Overcome by events, closing
